### ISSUE TITLE
Error when opening a Ruby buffer with lsp correctly configured

### ISSUE THREAD

**Issue by reynard93** [oskarrrrrrr/symbols.nvim#8]
![image](https://github.com/user-attachments/assets/17e6a99c-232c-4218-834f-167b02c4d7aa)

running :LspInfo gives me this:
![image](https://github.com/user-attachments/assets/f22b1a08-b2e3-4d49-aa3c-64c7df9081ae)

treesitter also correctly set up, not sure what is the issue help appreciated. great work btw i like it more than aerial and have moved over but cannot resolve this issue

**Comment by oskarrrrrrr** [comment]
The problem is that `solograph` provides symbols in `SymbolInformation` format instead of `DocumentSymbol` format (see [lsp-doc](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_documentSymbol)). I knew that this was possible but none of the LSP servers I tested with so far used it and I decided to wait until I find one that does.

`SymbolInformation` is a bit worse format for us because it gives a flat list of symbols:

> SymbolInformation[] which is a flat list of all symbols found in a given text document. Then neither the symbol’s location range nor the symbol’s container name should be used to infer a hierarchy.

I should be able to fix it soon.

**Comment by reynard93** [comment]
you are welcome! 👍

**Comment by oskarrrrrrr** [comment]
Should be fixed: [06cef1b](https://github.com/oskarrrrrrr/symbols.nvim/commit/06cef1b15afa3f354be853ff185acbc884737736)

Looks like this:
<img width="878" alt="Screenshot 2024-11-28 at 01 40 12" src="https://github.com/user-attachments/assets/3c55918f-2e09-40ff-8057-2389c18c21a5">

Note that there is symbol nesting, the tree is built by  looking at symbol ranges.  In my limited testing it seems to be working.

Also added this idiomatic config for Ruby:
  ```lua
ruby = {
      Class = "class",
      Module = "module",
      Property = "attr",
      Constant = "const",
      Function = "def",
      Method = "def",
  }
```
